prompts/template_xml/cpp-specific-priming-filler.txt

```diff
@@ -1,41 +1,132 @@
 
-Use <code>FuzzedDataProvider</code> to generate these inputs. You MUST declare it in <code>LLVMFuzzerTestOneInput</code>, like this:
+<instruction>
+
+Use <code>FuzzedDataProvider</code> to generate these inputs, it is a single-header C++ library that is helpful for splitting a fuzz input into multiple parts of various types. It can be included via
+<code>
+#include <fuzzer/FuzzedDataProvider.h>
+</code>
+
+An advantage and disadvantage of using this library is that the input splitting happens dynamically, i.e. you don't need to define any structure of the input. This might be very helpful in certain cases, but would also make the corpus to be no longer in a particular format. For example, if you fuzz an image parser and split the fuzz input into several parts, the corpus elements will no longer be valid image files, and you won't be able to simply add image files to your corpus.
```

Comment:
Do we see improvements from this bit?

Reply:
No, I misread it and thought it was something useful.
I have rewrote the useful part as a guidelines in https://github.com/google/oss-fuzz-gen/pull/331/files#diff-731a2942b5258de053b8673a6f1b9fb37898dc05167f220508a2d62a412c32f7R36